### ISSUE TITLE
Issues with testbench - md5

### ISSUE THREAD

**Issue by temhem2018** [frysztak/md5#1]
Hi,

i've problems with the testbench.
I used a empty message for the test and i got wrong hash values.
Perhaps there is a problem with the calculation of the message length?
If i force the data_in to 32'h00000000 it calculates the hashsum correctly - but with a empty message ( only zeros and a message length of 32'h190 ) it doesn't.

Maybe you could help me with my issue.

Yours sincerely
Mehmet Cibir
m.cibir@hotmail.de

![grafik](https://user-images.githubusercontent.com/37616905/37702726-c0d720e2-2cf3-11e8-92cf-7bcb08b83bbb.png)

**Comment by frysztak** [comment]
Hi :)

As far as I remember it correctly, input message must be 512 bits long, as seen in signal `message` in MD5_tb.vhd. I know it's extremely inflexible, and basically renders this whole project unusable, but it was just a quick student project, and nothing serious.

Best regards,
Sebastian

**Comment by temhem2018** [comment]
Hi Sebastian,

thank you for your answer 😊
Yes, the Input message must be 512 bits long, thats true.
448 bit length for message+binary 1+zero’s + 64 bit with the value of message_length ( low-byte order first).

We looked again at the source code and the simulation in ModelSim. He does not calculate and transmit the message length correctly. We solved the problem with a new "message_signal" that we manually "pad".
For that, we had to rewrite the state_machine a bit.

By the way, we also only deal with that because we do a study project in vhdl, too.

The rest works really great!
Thank you very ver much that you make the effort and that you share your source code, which helped us a lot!!


Best regards,
Mehmet



Von: Sebastian Frysztak<mailto:notifications@github.com>
Gesendet: Donnerstag, 22. März 2018 18:48
An: sebastian-frysztak/md5<mailto:md5@noreply.github.com>
Cc: temhem2018<mailto:m.cibir@hotmail.de>; Author<mailto:author@noreply.github.com>
Betreff: Re: [sebastian-frysztak/md5] Issues with testbench - md5 (#1)


Hi :)

As far as I remember it correctly, input message must be 512 bits long, as seen in signal message in MD5_tb.vhd. I know it's extremely inflexible, and basically renders this whole project unusable, but it was just a quick student project, and nothing serious.

Best regards,
Sebastian
